src/main/java/com/aws/greengrass/clientdevices/auth/configuration/RuntimeConfiguration.java

```diff
@@ -66,4 +72,31 @@ public void updateCAPassphrase(String passphrase) {
         caPassphrase.withValue(passphrase);
     }
 
-}
\ No newline at end of file
+    /**
+     * Update (or create if not available) opaque client device thing details in the config.
+     *
+     * @param thingName    client device Thing name
+     * @param thingDetails opaque client device Thing details
+     */
+    public void updateClientDeviceThing(String thingName, Map<String, Object> thingDetails) {
+        Topics clientDeviceThings = config.lookupTopics(CLIENT_DEVICE_THINGS_KEY);
+        Topics clientDeviceThing = clientDeviceThings.lookupTopics(thingName);
+        clientDeviceThing.replaceAndWait(thingDetails);
```

Comment:
This can be expensive. I'd recommend only updating the parts that have actually changed.

Reply:
Right, agreed. I added new api to update only the changed thing details: 
```
updateClientDeviceThing(String thingName, Map<String, Object> thingDetails, String schemaVersion) {}
```
Let me create a separate PR where adapter figures out what details changed and calls this RuntimeConfig api accordingly ^